.github/workflows/run_tests.yml

```diff
@@ -16,8 +16,6 @@ jobs:
 
     steps:
```

Comment:
since you mentioned import cleanup etc. there are quite some tools to automatically keep the code pretty clean, see https://medium.com/semantixbr/how-to-make-your-code-shine-with-gitlab-ci-pipelines-48ade99192d1

e.g. flake can be automated like this https://github.com/esa/torchquad/blob/develop/.github/workflows/run_tests.yml#L42